src/containers/InvoicePage.js

```diff
@@ -1,51 +1,43 @@
-import React from 'react';
-import Immutable from 'immutable';
+import React, { Component } from 'react';
 import { connect } from 'react-redux';
-import { createInvoice } from '../thunks';
-import  CreateInvoiceForm from '../components/CreateInvoiceForm';
+import InvoiceTable from '../components/InvoiceTable';
+import { markInvoicePaid } from '../thunks'
 
-function InvoicePage({clients, users, onCreateClick}) {
-  return (
-    <CreateInvoiceForm onSubmit={onCreateClick}
-                         clients={clients}
-                         users={users}/>
-  );
-}
+class InvoicePage extends Component {
+  constructor(props) {
+    super(props);
 
-function clientFromProject(project) {
-  const name = project.get('name');
-  return name.split('|', 2)[0];
-}
+    this.onInvoiceTableSubmit = this.onInvoiceTableSubmit.bind(this);
+  }
 
-function mapStateToProps(state) {
-  const projects = state.getIn(['entities', 'projects']);
-  const users = state.getIn(['entities', 'users']);
-  if (!projects) {
-    return {
-      clients: Immutable.List([])
-    };
+  onInvoiceTableSubmit(invoiceId) {
+    this.props.dispatch(markInvoicePaid(invoiceId));
+    return;
+  }
+
+  render() {
+    const savedInvoice = this.props.entities
+      .get('invoices')
+      .get(parseInt(this.props.params.invoiceId, 10));
+
+    if (this.props.isFetching || !savedInvoice) {
```

Comment:
Because this feature doesn't work when we navigate to the URL directly anyway, I think this code should go away. It is not adding any value.